source/postprocess/plateness_statistics.cc

```diff
@@ -0,0 +1,182 @@
+/*
+  Copyright (C) 2011-2021 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file doc/COPYING.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+
+#include <aspect/simulator.h>
+#include <aspect/utilities.h>
+#include <aspect/postprocess/plateness_statistics.h>
+#include <aspect/global.h>
+#include <aspect/geometry_model/spherical_shell.h>
+
+#include <deal.II/base/quadrature_lib.h>
+#include <deal.II/fe/fe_values.h>
+
+
+namespace aspect
+{
+  namespace Postprocess
+  {
+    template <int dim>
+    std::pair<std::string,std::string>
+    PlatenessStatistics<dim>::execute (TableHandler &statistics)
+    {
+      const types::boundary_id top_boundary_id = this->get_geometry_model().translate_symbolic_boundary_name_to_id("top");
+
+      // create a quadrature formula for the velocity.
+      const QGauss<dim-1> quadrature_formula (this->introspection().polynomial_degree.velocities+1);
+   
+      FEFaceValues<dim> fe_face_values (this->get_mapping(),
+                                        this->get_fe(),
+                                        quadrature_formula,
+                                        update_gradients |
+                                        update_quadrature_points |
+                                        update_JxW_values);
+         
+      std::vector<Tensor<1,dim>> velocities (fe_face_values.n_quadrature_points);
+
+      // Vector to store the integral of the second invariant of the strain rate
+      // over the surface.
+      double local_second_invariant_of_strain_rate_integral = 0.0;
+      double local_surface_area_integral = 0.0;
+
+      std::vector<Tensor<1,dim>> local_second_invariant_of_strain_rate(fe_face_values.n_quadrature_points);
+      std::vector<Tensor<1,dim>> local_surface_area(fe_face_values.n_quadrature_points); 
```

Comment:
Your error is due to the fact that this should probably be
```suggestion
      std::vector<double> local_surface_area(fe_face_values.n_quadrature_points); 
```

Reply:
Yes, thank you:)